pyptex/__init__.py

```diff
@@ -229,298 +241,305 @@ def genname(self,pattern="fig{gencount}.eps"):
         `pyp.genname()` does not actually create the file. `pyp.genname()` increments
         `pyp.gencount` every time it is called.
         """
-        self.gencount = self.gencount+1
-        return self.gendir+'/'+pattern.format(**self.__dict__)
+        self.gencount += 1
+        return f'{self.gendir}/{pattern.format(**self.__dict__)}'
 
-    def savefig(self,fig,pattern="fig{gencount}.eps",**kwargs):
+    def savefig(self, fig, pattern: str = 'fig{gencount}.eps', **kwargs):
         """Save a figure to the a-generated/* subdirectory.
-        
+
         If `pyp` is an object of type `pyptex`:
-        `pyp.savefig(fig)` saves a SymPy or matplotlib figure to the `a-generated/*` 
-        subdirectory, using the `pyptex.genname()` automatically generated filename. 
-        `pyp.savefig(fig,pattern)` further specifies the filename pattern of the
+        `pyp.savefig(fig)` saves a SymPy or matplotlib figure to the `a-generated/*`
+        subdirectory, using the `pyptex.genname()` automatically generated filename.
+        `pyp.savefig(fig, pattern)` further specifies the filename pattern of the
         generated name, see `genname()`.
-        `pyp.savefig(fig,pattern,...)` passes any further keyword arguments directly
+        `pyp.savefig(fig, pattern, ...)` passes any further keyword arguments directly
         to the `savefig` function from matplotlib.
 
         A typical way of using this from a TeX file is:
         `\\includegraphics{@{pyp.savefig(...)}}`
         """
-        if(self.__sympy_plot__ == None):
-            self.__sympy_plot__ = sympy.plotting.plot(1,show=False).__class__
+        if self.__sympy_plot__ is None:
+            self.__sympy_plot__ = sympy.plotting.plot(1, show=False).__class__
         figname = self.genname(pattern)
-        if(fig.__class__==self.__sympy_plot__):
+        if fig.__class__ == self.__sympy_plot__:
             backend = fig.backend(fig)
             backend.process_series()
             backend.fig.savefig(figname, **kwargs)
         else:
             fig.savefig(figname, **kwargs)
         self.dep(figname)
         return figname
+
     def generateddir(self):
-        """This is an internal function that creates the generated directory"""
-        self.gendir = self.filename+'-generated'
+        """This is an internal function that creates the generated directory."""
+        self.gendir = f'{os.fspath(self.filename)}-generated'
         if not os.path.exists(self.gendir):
             os.makedirs(self.gendir)
         self.gencount = 0
 
-    def __init__(self,texfilename,argv=[],latexcommand=False):
+    def __init__(self, texfilename, argv: List = None, latexcommand: bool = False):
```

Comment:
This is the "bug" I thought I found, but now that I think of it, I think it says somewhere that type labels don't affect execution so I guess this isn't a bug-bug, but I think the type information here is misleading?

Your new version, I think, is slightly incorrect. The main glitch is with "latexcommand" but I start with argv. It's true I didn't do it anywhere, but in the back of my mind I thought that argv, despite its name, could be any desired thing you want to pass into the child scope. Users might enjoy being able to do pyptex('b.tex',17) instead of wrapping it in brackets.

Also, why did you change the default value from [] to None? The real default value as used is [], what's gained by this kind of "indirection"?

The main thing is latexcommand, which is meant to be either a string or a bool. If latexcommand is falsy, no shell command is executed at the end of compilation. if latexcommand is truthy, it had better be a string and it's formatted and given to os.system, the purpose is to call pdflatex or similar.
```suggestion
    def __init__(self, texfilename, argv = [], latexcommand = False):
```

Reply:
Hi, I'm back. For your first comment (up at the top): I've run the tests and made sure everything works, and I'm less of GitHub expert than you, so I have no idea. I don't think so, about the linter. though.

Yeah, I was definitely to gung-ho about type annotations, and here I definitely added them without fully understanding the context. I'll remove those.

As for the `None` (for [this comment, too](https://github.com/sloisel/pyptex/pull/2#discussion_r389358773)), a list as a default value in arguments is considered global, and will be resued with each instance of the class. For example:

```python3
>>> class C:
...     def __init__(self, a=[]):
...             self.a = a
...
>>> instance1 = C()
>>> instance1.a.append(1)
>>> instance2 = C()
>>> instance2.a
[1]
```
The `None` and `[] if argv is None else []` gets around that by creating the list in the local scope.

Reply:
Oh ok, I did not know that. Definitely sounds like a bug on my part, the reused default argument thing.